app/scripts/lib/url.js

```diff
@@ -105,6 +99,21 @@ define(function (require, exports, module) {
       }
       _.extend(params, newParams);
       return uri + this.objToSearchString(params);
+    },
+
+    /**
+     * Clean the search string by only allowing search parameters declared in
+     * `allowedFields`
+     *
+     * @param {String} uri - uri with search string to clean.
+     * @param {String[]} allowedFields - list of allowed fields.
+     * @returns {String}
+     */
+    cleanSearchString (uri, allowedFields) {
+      const [ base, search ] = uri.split('?');
+      const cleanedQueryParams =
+        this.searchParams(search, allowedFields);
```

Comment:
Great catch. `searchParam`'s fallback behavior to import from window.location.search is a total footgun. We always pass in what are supposed to be the search parameters already, so I'm going to remove that.